include/emp/prefab/QueueManager.hpp

```diff
@@ -0,0 +1,331 @@
+ * 
+ * Example of adding metrics:
+ * 
+ * my_queue_manager.AddMetric([](){return emp::to_string(std::chrono::system_clock::now());}, "Wall time");
+ * 
+ * Once a QueueManager has all the metrics set up, a button and text input for queueing runs and a table for 
+ * displaying progress can be added to a web page. To add the button/text input for queueing, the QueueManager
+ * needs to know two things: where to get the SettingConfig for the newly-queued runs, and how to figure out 
+ * how many epochs (time steps) to run them for. Both can be specified with functions.
+ * 
+ * Example:
+ * 
+ * std::function<emp::SettingConfig()> get_setting_config = [](){
+ *      emp::SettingConfig my_settings;
+ *      my_settings.AddSetting<double>("my_param") = {.9};
+ *      return my_settings;
+ * };
+ * std::function<int()> get_epochs = [](){return 50;}; // Always run for 50 epochs
+ * 
+ * emp::web::Document doc("emp_base");
+ * my_queue_manager.AddQueueButton(get_setting_config, get_epochs); // Add button and text input
+ * my_queue_manager.BuildTable(); // Add progress table
+ * doc << my_queue_manager.GetDiv(); // Get the div storing QueueManager stuff and put it on the document
+ * 
+ * Once your QueueManager is set up, you'll need integrate it with the rest of your code so that it
+ * appropriately pulls new runs from the queue when appropriate, does set up for each run at the beginning,
+ * and updates the current epoch and table when appropriate. The way to do this will vary based on the rest
+ * of your code. Assuming each run of your code involves a loop that occurs for a certain number of time
+ * steps (which correspond to epochs in the QueueManger), here is a template for how you might set up
+ * the body of that loop:
+ * 
+ * if (!run_list.IsEmpty()) { // If there isn't stuff in the queue, support running program normally
+ *   emp::QueueManager::RunInfo & run = run_list.FrontRun();
+ *   if (run.GetEpoch() == 0) {  // Are we starting a new run?
+ *     // Add beginning of run set up here
+ *     // e.g. load in parameter settings from run.GetConfig()
+ *     // If you want to do an initial visualization of something,
+ *     // that goes here too.
+ *     // Make sure to fully clean up from any previous run
+ *   } 
+ *   run.IncEpoch();  // Assumes the loop this is inside runs once per time step
+ * }
+ * 
+ * // Add the rest of the work for an individual time step of your code here
+ * // e.g. update the world, draw stuff, etc.
+ *
+ * if (!run_list.IsEmpty()) { // If we're running something from the queue, update the table
+ *   run_list.Update();       // and check whether we have finished the current job
+ * }
+ * 
+ * For an example of this template in action, see the SpatialCoop2017 demo.
+ * 
+ * More information is available in this blog post: https://mmore500.com/waves/blog/queuemanager.html (this 
+ * tool was written by Juan Chavez as part of WAVES 2020)
+ */
+
+#pragma once
```

Comment:
Switch this out for #ifndef/#define style header guards, I think we're trying to stay /entirely/ standards-compliant